drivers/eeprom/eeprom_mb85rcxx.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT fujitsu_mb85rcxx
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/eeprom.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(EEPROM_MB85RCXX, CONFIG_EEPROM_LOG_LEVEL);
+
+struct mb85rcxx_config {
+	struct i2c_dt_spec i2c;
+	struct gpio_dt_spec wp_gpio;
+	size_t size;
+	size_t pagesize;
+	uint8_t addr_width;
+	bool readonly;
+};
+
+struct mb85rcxx_data {
+	struct k_mutex lock;
+};
+
+static int mb85rcxx_write_protect_set(const struct device *dev, int value)
+{
+	const struct mb85rcxx_config *cfg = dev->config;
+
+	if (!cfg->wp_gpio.port) {
+		return 0;
+	}
+
+	return gpio_pin_set_dt(&cfg->wp_gpio, value);
+}
+
+static uint16_t mb85rcxx_translate_address(const struct device *dev, off_t offset, uint8_t *addr)
+{
+	const struct mb85rcxx_config *cfg = dev->config;
+
+	off_t page_offset = offset % cfg->pagesize;
+
+	if (cfg->addr_width > 8) {
+		sys_put_be16(page_offset, addr);
+		addr[0] &= BIT_MASK(cfg->addr_width - 8);
+	} else {
+		addr[0] = page_offset & BIT_MASK(cfg->addr_width);
+	}
+
+	return cfg->i2c.addr + (offset >> cfg->addr_width);
+}
+
+static size_t mb85rcxx_remaining_len_in_page(const struct device *dev, off_t offset, size_t len)
+{
+	const struct mb85rcxx_config *cfg = dev->config;
+	off_t page_offset = offset % cfg->pagesize;
+	size_t rem = cfg->pagesize - page_offset;
+
+	if (rem > len) {
+		rem = len;
+	}
+
+	return rem;
+}
+
+static int mb85rcxx_init(const struct device *dev)
+{
+	const struct mb85rcxx_config *cfg = dev->config;
+	struct mb85rcxx_data *data = dev->data;
+
+	k_mutex_init(&data->lock);
+
+	if (!i2c_is_ready_dt(&cfg->i2c)) {
+		LOG_ERR("i2c bus device not ready");
+		return -EINVAL;
+	}
+
+	if (cfg->wp_gpio.port) {
+		if (!gpio_is_ready_dt(&cfg->wp_gpio)) {
+			LOG_ERR("wp gpio device not ready");
+			return -EINVAL;
+		}
+
+		int err = gpio_pin_configure_dt(&cfg->wp_gpio, GPIO_OUTPUT_ACTIVE);
+
+		if (err) {
+			LOG_ERR("failed to configure WP GPIO pin (err %d)", err);
+			return err;
+		}
+	}
+	return 0;
+}
+
+static int mb85rcxx_read(const struct device *dev, off_t offset, void *buf, size_t len)
+{
+	const struct mb85rcxx_config *cfg = dev->config;
+	struct mb85rcxx_data *data = dev->data;
+	uint8_t addr[2];
+	uint16_t i2c_addr = 0;
+	size_t len_in_page = 0;
+	int ret = 0;
```

Comment:
These don't need to be initialized, drop the `= 0`

Reply:
Thanks, I removed the `= 0`